src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.removeEventListener('click', resolve);
+  };
+
+  document.addEventListener('click', handler);
+
+  setTimeout(reject, 3000);
+});
+
+const secondPromise = new Promise((resolve, reject) => {
+  const handler = () => {
+    resolve();
+    document.removeEventListener('click', resolve);
+    document.removeEventListener('contextmenu', resolve);
+  };
+
+  document.addEventListener('contextmenu', handler);
+  document.addEventListener('click', handler);
+});
+
+const thirdPromise = new Promise((resolve, reject) => {
+  let left = false;
+  let right = false;
+
+  const leftClick = () => {
+    left = true;
+
+    if (right) {
+      resolve();
+      document.removeEventListener('click', leftClick);
+      document.removeEventListener('contextmenu', rightClick);
```

Comment:
Bug: You are removing the event listener with 'rightClick' as a handler, but you added the listener with 'contextmenu' event. This will not remove the listener as expected.